src/geometry.rs

```diff
@@ -1,20 +1,405 @@
+            self.check_vertex_attribute(slot)?;
+        }
+        match slice {
+            BufferSlice::Buffer {
+                buffer,
+                offset,
+                size,
+            } => {
+                dbg!(
+                    "Binding buffer slice to slot {}, offset {}, stride {}, count {}",
+                    slot,
+                    offset,
+                    stride,
+                    count
+                );
+                let bindings = self.attachments.entry(usage).or_insert_with(Vec::new);
+                match bindings.iter().position(|a| a.slot == slot) {
+                    // If the slot is already bound, remove the old binding and
+                    // replace it with the new one.
+                    Some(i) => {
+                        bindings.remove(i);
+                        unsafe {
+                            rtcSetGeometryBuffer(
+                                self.handle,
+                                usage,
+                                slot,
+                                format,
+                                buffer.handle,
+                                offset,
+                                stride,
+                                count,
+                            )
+                        };
+                        bindings.push(AttachedBuffer {
+                            slot,
+                            source: BufferSlice::Buffer {
+                                buffer,
+                                offset,
+                                size,
+                            },
+                            format,
+                            stride,
+                        });
+                        Ok(())
+                    }
+                    // If the slot is not bound, just bind the new buffer.
+                    None => {
+                        unsafe {
+                            rtcSetGeometryBuffer(
+                                self.handle,
+                                usage,
+                                slot,
+                                format,
+                                buffer.handle,
+                                offset,
+                                stride,
+                                count,
+                            )
+                        };
+                        bindings.push(AttachedBuffer {
+                            slot,
+                            source: BufferSlice::Buffer {
+                                buffer,
+                                offset,
+                                size,
+                            },
+                            format,
+                            stride,
+                        });
+                        Ok(())
+                    }
+                }
+            }
+            BufferSlice::Internal { .. } => {
+                eprint!("Internally managed buffer cannot be shared!");
```

Comment:
I think we should re-word this slightly because "internally managed buffer" could also be read as buffers made via `Buffer::new` which are also memory "internally managed" by Embree.

Maybe `BufferSlice::Internal` could be `BufferSlice::GeometryLocal` ? Then the error here would be about trying to share a geometry-local buffer.

Reply:
Renamed as `GeometryLocal`